### ISSUE TITLE
Powder Ability Bar flickers due to action bar changing

### ISSUE THREAD

**Issue by kristofbolyai** [Wynntils/Wynntils#930]
This bug might be relevant in all action bar handlers. Should we do something in the handlers to handle this, or leave it to the implementations?

**Comment by magicus** [comment]
This might or might not be a regression. As it was coded before, the powder special bar was reset to 0 when the coordinates segment was displayed, but not when any other segment (sprint, spell cast) was displayed.

Maybe that was a useful heuristic. The coordinate segment is kind of the "fallback" one, indicating that no other indication is needed. Otoh, I assume that you would like have a powder special charge at 100%, start running, and quickly shift-click your powder special, and possibly the action bar will not be updated fast enough to catch this.

So basically there's two ways to do this:
1) play it safe -- only show what is really shown on the action bar at the time
2) add some heuristics -- may make for a better experience, but may also be wrong

**Comment by kristofbolyai** [comment]
Yes, this is a regression. It was only cleared by coordinates, which is almost enough, adding clearing on slot switching would solve almost every issue I think.

**Comment by magicus** [comment]
I was already thinking about 2 for sprinting; we should be able to detect if we are still in sprint mode and if so, extrapolate the sprint bar, even if we have started casting a spell.

Powder special usage appears as a Subtitle, so we can track if it's been used, and assume the charge is at the same level until we see the special subtitle, or we switch weapon (or if there is anything else that can make us lose the charge).

**Comment by magicus** [comment]
Okay. Let's treat this as a regression that needs fixing. I'm feeling slightly better now, so I can give it a shot.

**Comment by kristofbolyai** [comment]
> I was already thinking about 2 for sprinting; we should be able to detect if we are still in sprint mode and if so, extrapolate the sprint bar, even if we have started casting a spell.
> 
> 
> 
> Powder special usage appears as a Subtitle, so we can track if it's been used, and assume the charge is at the same level until we see the special subtitle, or we switch weapon (or if there is anything else that can make us lose the charge).

Don't think there's anything else (apart from world switching and other obvious stuff)
